Fixes #51 Fixes #55 - Added BasicTilemapTestSuite

Indibulous opened the pull request:
### Description

Created BasicTilemapTestSuite, a TestSuite containing 24 facts/testAsyncs that test all the required functionality listed in https://github.com/YoYoGames/GM-TestFramework/issues/51: 
- tilemap_create_exists_destroy
- layer_tilemap_get_id
- tilemap_set_get_mask
- tilemap_set_get_global_mask
- tilemap_set_get_tileset
- tilemap_get_frame
- tilemap_get_tile_width
- tilemap_get_tile_height
- tilemap_set_get_width
- tilemap_set_get_height
- tilemap_set_get_x
- tilemap_set_get_y
- tilemap_set_get
- tilemap_set_get_at_pixel
- tilemap_get_cell_x_at_pixel
- tilemap_get_cell_y_at_pixel
- tilemap_clear
- draw_tilemap
- tile_set_get_empty
- tile_set_get_index
- tile_set_get_flip
- tile_set_get_mirror
- tile_set_get_rotate
- draw_tile

Some other resources required for the tests have also been added: 
- tilesAutoAnimated
- tilesAutoReplacement
- drawTilemapTestExpected.png
- drawTilemapTestExpectedBuffer
- drawTileTestExpected.png
- drawTileTestExpectedBuffer

And a new object to be used with testAsync for testing draw functions was added, implementing the functionality described in https://github.com/YoYoGames/GM-TestFramework/issues/55: 
- objTestAsyncDraw

Indibulous committed:
Added BasicTilemapTestSuite

Created BasicTilemapTestSuite, a TestSuite containing 24 facts/testAsyncs that test all the required functionality:
tilemap_create_exists_destroy
layer_tilemap_get_id
tilemap_set_get_mask
tilemap_set_get_global_mask
tilemap_set_get_tileset
tilemap_get_frame
tilemap_get_tile_width
tilemap_get_tile_height
tilemap_set_get_width
tilemap_set_get_height
tilemap_set_get_x
tilemap_set_get_y
tilemap_set_get
tilemap_set_get_at_pixel
tilemap_get_cell_x_at_pixel
tilemap_get_cell_y_at_pixel
tilemap_clear
draw_tilemap
tile_set_get_empty
tile_set_get_index
tile_set_get_flip
tile_set_get_mirror
tile_set_get_rotate
draw_tile

Some other resources required for the tests have also been added:
tilesAutoAnimated
tilesAutoReplacement
drawTilemapTestExpected.png
drawTilemapTestExpectedBuffer
drawTileTestExpected.png
drawTileTestExpectedBuffer

And a new object to be used with testAsync for testing draw functions was added:
objTestAsyncDraw

Indibulous committed:
Improved comments & a couple small fixes

Improved and added comments throughout BasicTilemapTestSuite and objTestAsyncDraw, as well as fixed issues in the tile_set_get_mirror andtilemap_get_frame facts, and removed some redundant code in the tilemap_set_get_global_mask Fact

DiasFranciscoA commented on projects/xUnit/objects/objTestAsyncDraw/Create_0.gml:
```diff
@@ -0,0 +1,97 @@
+/// @description Begin running test, and declare some variables and functions for draw buffer comparisons
+// The test name to use as a prefix for any saved/loaded files during the test.
+testFileName = ""
```
we can probably refactor this out from the test handler and put it into some reusable space maybe... "utils" ?

Indibulous committed:
Draw comparison functions moved to draw_test_utils

The functions start_draw_comparison() and end_draw_comparison() (formerly StartDrawComparison() and EndDrawComparison()) have been moved into a script called draw_test_utils and been modified to use no instance variables, as well as matching the formatting of the other util scripts.
